### PR TITLE
update requirements for frontend

### PR DESCRIPTION
**Pull request by alexandrasp**
It removes requirements that front-end application is not
using to promote a cleaner environment.

Signed-off-by: Alexandra Pereira <alexandra.pereira@collabora.com>

### CODE REVIEW

**Comment by gctucker** [comment]
Thanks, looks fine based on the fact that these Python packages aren't used anywhere in the code.

@mgalka @khilman If this is something you can test, does it work for you with a fresh Buster install in a VM or Docker container?

**Comment by khilman** [comment]
I get this in a fresh docker install using the staging branch (master branch is fine):
```
Tue Dec 10 22:35:48 2019 - *** Operational MODE: preforking+threaded ***                          
/usr/local/lib/python2.7/dist-packages/flask_cache/__init__.py:152: UserWarning: Flask-Cache: CACHE_TYPE is set to null, caching is effectively disabled.                                                                 
  warnings.warn("Flask-Cache: CACHE_TYPE is set to null, "                                        
/usr/local/lib/python2.7/dist-packages/flask_cache/jinja2ext.py:33: ExtDeprecationWarning: Importing flask.ext.cache is deprecated, use flask_cache instead.                                                              
  from flask.ext.cache import make_template_fragment_key                                          
Traceback (most recent call last):                                                                                                                                                                                        
  File "./server.py", line 21, in <module>                                                        
    from dashboard import app                                                                                                                                                                                             
  File "./dashboard/__init__.py", line 107, in <module>                                           
    import dashboard.utils.route as route                                                                                                                                                                                 
  File "./dashboard/utils/route.py", line 36, in <module>                                         
    import dashboard.utils.feed.job as jobfeed                                                                                                                                                                            
  File "./dashboard/utils/feed/__init__.py", line 19, in <module>                                 
    import bson                                                                                                                                                                                                           
ImportError: No module named bson                                                          
```

File: requirements.txt [hunk 1]
```diff
@@ -2,7 +2,4 @@ Flask-Cache
 Flask-WTF==0.11
 Flask==1.0
 redis==2.10.5
-hiredis
 requests
-simplejson
-pymongo==2.8.1
```
Comment (gctucker) [inline requirements.txt:8]: The `pymongo` package provides `bson`: https://pypi.org/project/pymongo/

Please leave `pymongo` in the list of requirements as it's needed.  It should also probably be aligned with the version used in the backend (3.9.0) with a new PR.

**Review by gctucker — CHANGES_REQUESTED** [review]

**Comment by gctucker** [comment]
> I get this in a fresh docker install using the staging branch (master branch is fine):
> 
> ```
> Tue Dec 10 22:35:48 2019 - *** Operational MODE: preforking+threaded ***                          
> /usr/local/lib/python2.7/dist-packages/flask_cache/__init__.py:152: UserWarning: Flask-Cache: CACHE_TYPE is set to null, caching is effectively disabled.                                                                 
>   warnings.warn("Flask-Cache: CACHE_TYPE is set to null, "                                        
> /usr/local/lib/python2.7/dist-packages/flask_cache/jinja2ext.py:33: ExtDeprecationWarning: Importing flask.ext.cache is deprecated, use flask_cache instead.                                                              
>   from flask.ext.cache import make_template_fragment_key                                          
> Traceback (most recent call last):                                                                                                                                                                                        
>   File "./server.py", line 21, in <module>                                                        
>     from dashboard import app                                                                                                                                                                                             
>   File "./dashboard/__init__.py", line 107, in <module>                                           
>     import dashboard.utils.route as route                                                                                                                                                                                 
>   File "./dashboard/utils/route.py", line 36, in <module>                                         
>     import dashboard.utils.feed.job as jobfeed                                                                                                                                                                            
>   File "./dashboard/utils/feed/__init__.py", line 19, in <module>                                 
>     import bson                                                                                                                                                                                                           
> ImportError: No module named bson                                                          
> ```

This is because `pymongo` is actually required.

**Comment by gctucker** [comment]
Thanks.

@khilman Could you please check this now works as expected with your Docker container?

**Comment by khilman** [comment]
I now get this when starting the frontend in a fresh container:
```
[uWSGI] getting INI configuration from /srv/uwsgi.ini
Fri Feb  7 02:42:06 2020 - *** Starting uWSGI 2.0.18-debian (64bit) on [Fri Feb  7 02:42:06 2020] ***
Fri Feb  7 02:42:06 2020 - compiled with version: 8.2.0 on 10 February 2019 02:42:46
Fri Feb  7 02:42:06 2020 - os: Linux-4.19.0-0.bpo.5-amd64 #1 SMP Debian 4.19.37-5+deb10u2~bpo9+1 (2019-08-16)
Fri Feb  7 02:42:06 2020 - nodename: be2a5e4c7dbf
Fri Feb  7 02:42:06 2020 - machine: x86_64
Fri Feb  7 02:42:06 2020 - clock source: unix
Fri Feb  7 02:42:06 2020 - pcre jit enabled
Fri Feb  7 02:42:06 2020 - detected number of CPU cores: 40
Fri Feb  7 02:42:06 2020 - current working directory: /srv/kernelci-frontend/app
Fri Feb  7 02:42:06 2020 - detected binary path: /usr/bin/uwsgi-core
Fri Feb  7 02:42:06 2020 - setgid() to 33
Fri Feb  7 02:42:06 2020 - setuid() to 33
Fri Feb  7 02:42:06 2020 - chdir() to /srv/kernelci-frontend/app
Fri Feb  7 02:42:06 2020 - your memory page size is 4096 bytes
Fri Feb  7 02:42:06 2020 - detected max file descriptor number: 4096
Fri Feb  7 02:42:06 2020 - lock engine: pthread robust mutexes
Fri Feb  7 02:42:06 2020 - thunder lock: enabled
Fri Feb  7 02:42:06 2020 - uwsgi socket 0 bound to TCP address 0.0.0.0:5000 fd 3
Fri Feb  7 02:42:06 2020 - Python version: 2.7.16 (default, Oct 10 2019, 22:02:15)  [GCC 8.3.0]
Fri Feb  7 02:42:07 2020 - Python main interpreter initialized at 0x55c222550b40
Fri Feb  7 02:42:07 2020 - python threads support enabled
Fri Feb  7 02:42:07 2020 - your server socket listen backlog is limited to 128 connections
Fri Feb  7 02:42:07 2020 - your mercy for graceful operations on workers is 60 seconds
Fri Feb  7 02:42:07 2020 - mapped 703600 bytes (687 KB) for 8 cores
Fri Feb  7 02:42:07 2020 - *** Operational MODE: preforking+threaded ***
Traceback (most recent call last):
  File "./server.py", line 18, in <module>
    from dashboard import app
  File "./dashboard/__init__.py", line 41, in <module>
    from flask_wtf.csrf import (
  File "/usr/local/lib/python2.7/dist-packages/flask_wtf/__init__.py", line 17, in <module>
    from .recaptcha import *
  File "/usr/local/lib/python2.7/dist-packages/flask_wtf/recaptcha/__init__.py", line 2, in <module>
    from .fields import *
  File "/usr/local/lib/python2.7/dist-packages/flask_wtf/recaptcha/fields.py", line 4, in <module>
    from .validators import Recaptcha
  File "/usr/local/lib/python2.7/dist-packages/flask_wtf/recaptcha/validators.py", line 9, in <module>
    from werkzeug import url_encode
ImportError: cannot import name url_encode
```

**Comment by khilman** [comment]
If you change `from werkzeug import url_encode` to `from werkzeug.urls import url_encode` it works, but that's part of the flask-wtf package, so there's some dependency issue going on here.

Are you testing this on debian/buster?

**Comment by khilman** [comment]
OK, on buster, I had to add `werkzeug==0.16` to `requirements.txt` to use an older version that works with the version of flask-wtf that gets installed.

**Comment by khilman** [comment]
OK, my issue above is not related to this PR, but just to installing on buster.  I've created a separate PR for my issue: https://github.com/kernelci/kernelci-frontend/pull/69

With the pymongo added, this change is working for me, so no remaining objections from me to merge this.

**Review by gctucker — APPROVED** [review]

**Comment by gctucker** [comment]
Great, thanks @khilman for the review and testing.

**Pull request merged**